### PR TITLE
Support new log safety declarations

### PR DESCRIPTION
**Pull request by sfackler**
## Before this PR
We only supported detection of safe loggable endpoint parameters via the old `Safe` marker or `safe` tag.

## After this PR

==COMMIT_MSG==
Code generation now supports Conjure's new log safety declarations to track safe loggable endpoint parameters.
==COMMIT_MSG==

### CODE REVIEW

File: conjure-codegen/src/context.rs [hunk 1]
```diff
@@ -1002,6 +1027,84 @@ impl Context {
         }
     }
 
+    fn type_log_safety(&self, ty: &Type) -> Option<LogSafety> {
```
Comment (sfackler) [inline conjure-codegen/src/context.rs:1030]: This logic should mirror what's defined in https://github.com/palantir/conjure-java/blob/develop/conjure-java-core/src/main/java/com/palantir/conjure/java/types/SafetyEvaluator.java.

File: conjure-codegen/src/context.rs [hunk 2]
```diff
@@ -1002,6 +1028,84 @@ impl Context {
...
+                .union_()
+                .iter()
+                .map(|f| {
+                    f.safety()
+                        .cloned()
+                        .or_else(|| self.type_log_safety(f.type_()))
+                })
+                // The unknown variant is unsafe to log, and we don't want log safety to vary based
+                // on the type generation configuration.
+                .fold(Some(LogSafety::Unsafe), |a, b| self.combine_safety(a, b)),
```
Comment (dtobin) [inline conjure-codegen/src/context.rs:1090]: Just acknowledging that this is a different decision than the Java implementation.

**Review by dtobin — APPROVED** [review]

File: conjure-codegen/src/context.rs (hunk 2 above)
Comment (sfackler) [reply conjure-codegen/src/context.rs:1090]: Ah - I misread the Java logic! I will update to start with None.

Comment (dtobin) [reply conjure-codegen/src/context.rs:1090]: It was actually changed after the initial implementation: https://github.com/palantir/conjure-java/commit/4b08523c3b62455544f7b0544f07ae74eee4a13c

**Review by dtobin — APPROVED** [review]

**Pull request merged**